SQL/Archive/17.1/2017-08-23_changing_psc_centerid_column_definition.sql

```diff
@@ -0,0 +1,43 @@
+WARNINGS;
+SELECT 'Droping foreign keys' as 'Step #1';
+ALTER TABLE candidate DROP FOREIGN KEY `FK_candidate_1`;
+ALTER TABLE examiners_psc_rel DROP FOREIGN KEY `FK_examiners_psc_rel_2`;
+ALTER TABLE issues DROP FOREIGN KEY `fk_issues_5`;
+ALTER TABLE notification_spool DROP FOREIGN KEY `FK_notification_spool_2`;
+ALTER TABLE session DROP FOREIGN KEY `FK_session_2`;
+ALTER TABLE user_psc_rel DROP FOREIGN KEY `FK_user_psc_rel_2`;
+
+SELECT 'Changing column definitions' as 'Step #2-A';
+ALTER TABLE candidate CHANGE `CenterID` `CenterID` integer unsigned NOT NULL;
+ALTER TABLE examiners_psc_rel CHANGE `centerID` `centerID` integer unsigned NOT NULL;
+ALTER TABLE issues CHANGE `centerID` `centerID` integer unsigned DEFAULT NULL;
+ALTER TABLE session CHANGE `CenterID` `CenterID` integer unsigned DEFAULT NULL;
+ALTER TABLE user_psc_rel CHANGE `CenterID` `CenterID` integer unsigned NOT NULL;
+ALTER TABLE psc CHANGE `CenterID` `CenterID` integer unsigned NOT NULL AUTO_INCREMENT;
```

Comment:
@xlecours at this stage the `users` table `CenterID` column has not yet been dropped since it requires a manual step (its commented out in the patch) so it will block this patch. it needs to be added to the foreign key list.

Ideally, your patch would go look for all foreign keys to `psc` col `CenterID` and dynamically removes and readds foreign keys ! this is because some projects (like CCNA) have personalised tables that use foreign keys

Good luck

Reply:
@xlecours same goes for examiers table which would be deprecated after some manual steps

Reply:
output at this point in the patch
```
mysql> SELECT    TABLE_NAME,COLUMN_NAME,CONSTRAINT_NAME, REFERENCED_TABLE_NAME,REFERENCED_COLUMN_NAME FROM   INFORMATION_SCHEMA.KEY_COLUMN_USAGE WHERE   REFERENCED_TABLE_SCHEMA = 'rida_ccna' AND   REFERENCED_COLUMN_NAME = 'CenterID';
+------------------+-------------+-----------------------+-----------------------+------------------------+
| TABLE_NAME       | COLUMN_NAME | CONSTRAINT_NAME       | REFERENCED_TABLE_NAME | REFERENCED_COLUMN_NAME |
+------------------+-------------+-----------------------+-----------------------+------------------------+
| examiners        | centerID    | FK_examiners_1        | psc                   | CenterID               |
| user_psc_rel_TMP | CenterID    | FK_user_psc_rel_TMP_2 | psc                   | CenterID               |
| users            | CenterID    | FK_users_1            | psc                   | CenterID               |
+------------------+-------------+-----------------------+-----------------------+------------------------+
3 rows in set (0.24 sec)

```